src/ensembl/production/metadata/service.py

```diff
@@ -364,6 +364,95 @@ def get_genome_by_name(metadata_db, ensembl_name, site_name, release_version):
+    # This is sqlalchemy's metadata, not Ensembl's!
+    md = db.MetaData()
+    session = Session(metadata_db, future=True)
+    dataset_type = db.Table('dataset_type', md, autoload_with=metadata_db)
+
+    ds_types = db.select(dataset_type.c.name.label('data_set_type')).distinct()
+    ds_types_results = session.execute(ds_types).all()
+    ds_types_list = [ds_type['data_set_type'] for ds_type in ds_types_results]
+
+    return ds_types_list
+
+
+def populate_datasets_info(data):
+    ds_info = {
+        'dataset_uuid': data['dataset_uuid'],
+        'dataset_name': data['dataset_name'],
+        'dataset_version': data['dataset_version'],
+        'dataset_label': data['dataset_label'],
+        'version': data['version']
+    }
+    return ds_info
+
+
+def get_datasets_by_uuid(metadata_db, genome_uuid, release_version):
+    # This is sqlalchemy's metadata, not Ensembl's!
+    md = db.MetaData()
+    session = Session(metadata_db, future=True)
+
+    # Reflect existing tables, letting sqlalchemy load linked tables where possible.
+    genome = db.Table('genome', md, autoload_with=metadata_db)
+    # genome = md.tables['genome'] ?
+    dataset = db.Table('dataset', md, autoload_with=metadata_db)
+    genome_dataset = db.Table('genome_dataset', md, autoload_with=metadata_db)
+    dataset_type = db.Table('dataset_type', md, autoload_with=metadata_db)
+    ensembl_release = db.Table('ensembl_release', md, autoload_with=metadata_db)
+
+    genome_select = db.select(
+        genome.c.genome_uuid,
+        dataset.c.dataset_uuid,
+        # The label here should be identical to what's in proto file
+        dataset_type.c.name.label('data_set_type'),
+        dataset.c.name.label('dataset_name'),
+        dataset.c.version.label('dataset_version'),
+        dataset.c.label.label('dataset_label'),
+        ensembl_release.c.version
+        ).select_from(
+            genome
+        ).join(genome_dataset).join(dataset) \
+        .join(dataset_type).join(ensembl_release) \
+        .where(genome.c.genome_uuid == genome_uuid) \
+        .where(genome_dataset.c.is_current == 1) \
+        .distinct()
+
+    if release_version > 0:
+        genome_select = genome_select.filter_by(version=release_version)
+    else:
+        # if the release is not specified, we return the latest by default
+        genome_select = genome_select.filter_by(is_current=1)
+
+    # print("SQL QUERY ===> ", str(genome_select))
+
+    datasets_results = session.execute(genome_select).all()
+    # print("len of datasets_results ===> ", len(datasets_results)) 
+
+    if len(datasets_results) > 0:
+        all_data_set_types = get_data_set_types(metadata_db)
+        ds_obj = {}
+        for result in datasets_results:
+            ds_type_result = result['data_set_type']
+            if ds_type_result in all_data_set_types:
+                # Populate the objects bottom up
+                datasets_info_list = populate_datasets_info(result)
```

Comment:
I find the variable name `datasets_info_list` confusing, because it's a dictionary, not a list.  I think something like `dataset_dict` would be better

Reply:
Thanks for pointing this out